metagraph/src/cli/server.cpp

```diff
@@ -141,15 +141,16 @@ std::string process_search_request(const std::string &received_message,
     config.num_top_labels = json.get("num_labels", config.num_top_labels).asInt();
     config.fast = json.get("fast", config.fast).asBool();
 
-    std::unique_ptr<graph::align::IDBGAligner> aligner;
+    std::unique_ptr<graph::align::DBGAlignerConfig> aligner_config;
     if (json.get("align", false).asBool()) {
-        aligner = build_aligner(anno_graph.get_graph(), config);
+        aligner_config.reset(new graph::align::DBGAlignerConfig(
+            initialize_aligner_config(anno_graph.get_graph(), config)
+        ));
 
         // the fwd_and_reverse argument in the aligner config returns the best of
         // the forward and reverse complement alignments, rather than both.
         // so, we want to prevent it from doing this
-        auto &aligner_config = const_cast<graph::align::DBGAlignerConfig&>(aligner->get_config());
-        aligner_config.forward_and_reverse_complement = false;
+        aligner_config->forward_and_reverse_complement = false;
```

Comment:
why do you want to prevent this?

Reply:
this flag enables a mode in the aligner which returns the best alignment from the forward and reverse complement, which is different from the behaviour of query mode that returns both

Reply:
Why is it set to `true` by default?

Reply:
hold on, it is actually set to `false`, so we don't need to reset it here, right?

Reply:
I'll add an assert in QueryExecutor instead, so we make sure the value is correct

Reply:
actually, you're right, we don't need this at all. I think at one point, `forward_and_reverse_complement` was being set when the `--fwd-and-reverse` flag was set, but that was changed so it gets triggered by a different flag

Reply:
so yea, this isn't needed anymore